src/reducers/entities/posts.js

```diff
@@ -161,16 +161,55 @@ function handleReceivedPosts(posts = {}, postsInChannel = {}, postsInThread = {}
 }
 
 export function removeUnneededMetadata(post) {
-    if (!post.metadata || (!post.metadata.emojis && !post.metadata.files && !post.metadata.reactions)) {
+    if (!post.metadata) {
         return post;
     }
 
     const metadata = {...post.metadata};
+    let changed = false;
 
     // These fields are stored separately
-    Reflect.deleteProperty(metadata, 'emojis');
-    Reflect.deleteProperty(metadata, 'files');
-    Reflect.deleteProperty(metadata, 'reactions');
+    if (metadata.emojis) {
+        Reflect.deleteProperty(metadata, 'emojis');
+        changed = true;
+    }
+
+    if (metadata.files) {
+        Reflect.deleteProperty(metadata, 'files');
+        changed = true;
+    }
+
+    if (metadata.reactions) {
+        Reflect.deleteProperty(metadata, 'reactions');
+        changed = true;
+    }
+
+    if (metadata.embeds) {
+        let embedsChanged = false;
+
+        const newEmbeds = metadata.embeds.map((embed) => {
+            if (embed.type !== 'opengraph') {
+                return embed;
+            }
+
+            const newEmbed = {...embed};
+            Reflect.deleteProperty(newEmbed, 'data');
+
+            embedsChanged = true;
+
+            return newEmbed;
+        });
+
+        if (embedsChanged) {
+            metadata.embeds = newEmbeds;
+            changed = true;
+        }
+    }
+
+    if (!changed) {
+        // Nothing changed
```

Comment:
I'm not going to bother going back and changing this, but this is really one of the most pointless comments I've ever made